### PR TITLE
feat(connectToggleRefinement): add support for array values

### PR DESCRIPTION
**Pull request by Kocal**
Hi :wave: 

This PR is a proposal for #2182 which add support for multiples values on `on`/`off` for the toggle refinement connector.

See #2182 for more information.

**Result**

It allow us to use the toggle refinement connector with multiple values:
```js
connectToggleRefinement()({
  attribute: 'whatever',
  on: ['my', 'values'],
  off: ['something', 'else']
});
```

and the toggle refinement widget:
```js
search.addWidget(instantsearch.widgets.toggleRefinement({
  container: '#toggle',
  attribute: 'whatever',
  on: ['my', 'values'],
  off: ['something', 'else']
}));
```

On [the playground](https://codesandbox.io/s/instantsearchjs-y5pyp), I've used:
```js
  toggleRefinement({
    container: '#toggle',
    attribute: 'brand',
    on: ['Sony', 'Incipio'],
  }),
```

and you can see it toggle both values.

What do you think?
Thanks!

### CODE REVIEW

File: src/connectors/toggle-refinement/connectToggleRefinement.js [hunk 1]
```diff
@@ -185,7 +206,9 @@ export default function connectToggleRefinement(renderFn, unmountFn = noop) {
       },
 
       render({ helper, results, state, instantSearchInstance }) {
-        const isRefined = helper.state.isDisjunctiveFacetRefined(attribute, on);
+        const isRefined =
```
Comment (Haroenv) [inline src/connectors/toggle-refinement/connectToggleRefinement.js:209]: similar to this, we also need to edit it in `getWidgetState`: ?instant_search%5BrefinementList%5D%5Bbrand%5D%5B0%5D=Sony&instant_search%5BrefinementList%5D%5Bbrand%5D%5B1%5D=Incipio

**Comment by Haroenv** [comment]
Relevant sandbox: https://codesandbox.io/s/instantsearchjs-mp4zs?file=/src/app.js

File: src/connectors/toggle-refinement/connectToggleRefinement.js (hunk 1 above)
Comment (Kocal) [reply src/connectors/toggle-refinement/connectToggleRefinement.js:209]: Oh, thank you!

**Comment by eunjae-lee** [comment]
> Relevant sandbox: [codesandbox.io/s/instantsearchjs-mp4zs?file=/src/app.js](https://codesandbox.io/s/instantsearchjs-mp4zs?file=/src/app.js)

I clicked "branch" and "Metra", so got this url:

https://mp4zs.csb.app/?instant_search%5BrefinementList%5D%5Bbrand%5D%5B0%5D=Sony&instant_search%5BrefinementList%5D%5Bbrand%5D%5B1%5D=Incipio&instant_search%5BrefinementList%5D%5Bbrand%5D%5B2%5D=Metra&instant_search%5Btoggle%5D%5Bbrand%5D=true

but opening the url won't get Metra selected.

**Comment by Haroenv** [comment]
That's probably since they're on the same attribute, which isn't something we officially support I think. If the widgets are mounted in the other order, the result would be correct. I don't think we can really find a solution for that, except if toggle refinement doesn't remove other refinements in `getWidgetSearchParameters`

**Comment by eunjae-lee** [comment]
You're right. I changed the order of the widgets, and the result is correct. Since it's a "toggle" refinement, I guess the current behavior probably makes sense. We can ship it and later get feedback.

**Pull request merged**